Slimeperson customization 2022 expansion pack (also refactors a bit of the code related to it)

mcmeiler opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds lizard tails, frills, spines and wings to slimeperson shapeshifting

Deleted most copy-paste code in favor of separate procs to handle the sprite assignment/removal
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

more variety = good, I think

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: lizard tails, frills, spines and wings (non-functional, nice try, powergamer) to slimeperson shapeshifting
refactor: made slimeperson shapeshifting code *slightly* less awful.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

mcmeiler committed:
part 2 electric boogaloo: de-snowflake edition

(maybe)

mcmeiler committed:
I forgot to remove this :skull:

mcmeiler committed:
minor stuff

Kapu1178 commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -972,6 +972,9 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 				continue
+			if(isjellyperson(H))
+				S.image_alpha = 150 //Jellypeople sprite handling
+				S.hasinner = FALSE
```
Snowflake shit

Kapu1178 commented on code/modules/mob/living/carbon/human/species_types/jellypeople.dm:
```diff
@@ -171,44 +171,97 @@
+					if("Ears")
+						handle_sprites(lowertext(selected_part), H, GLOB.ears_list)
+					if("Tails") //Tails are a special case since they're separate from each other, and go into different features and mutant_bodyparts.
+						var/datum/sprite_accessory/tails/SA = new(src) // I *could* move this to a separate proc, but I don't see the point. It'd only be called this one time.
```
dont instantiate an object, just use initial()

Kapu1178 commented on code/modules/mob/living/carbon/human/species_types/jellypeople.dm:
```diff
@@ -171,44 +171,97 @@
+							H.dna.species.mutant_bodyparts |= "tail_human"
+							H.dna.features["tail_human"] = SA.name
+							H.update_body()
+							qdel(SA)
```
wow thats alot of copy paste

Kapu1178 commented on code/modules/mob/living/carbon/human/species_types/jellypeople.dm:
```diff
@@ -171,44 +171,97 @@
+		H.dna.features["[sprite]"] = SA.name
+		H.update_body()
+		qdel(SA)
+		return
```
alot more copy paste. thin this out.

mcmeiler committed:
attempt uno at addressing review.

Copy-paste code turned into procs. Alpha assignment somewhat de-snowflaked. Couldn't find a better indicator than hair_alpha for how transparent a species is.

mcmeiler commented:
Go ahead and yell at me all you want, I'm going to bed. This still sucks, but it's no longer copy-paste.

Kapu1178 commented on code/modules/mob/living/carbon/human/species_types/jellypeople.dm:
```diff
@@ -171,44 +171,94 @@
+				//Mutant Bodyparts
+
+		if("Mutant Bodyparts")
+			var/list/part_list = list("Ears", "Tails", "Frills", "Spines", "Wings") //Selection, if you want more stuff added, do it yourself
+			var/selected_part = input(owner, "Which bodypart do you want to alter?", "Mutant Bodypart Alteration") as null|anything in part_list + "Cancel"
+			if(selected_part)
+				if(selected_part == "Cancel")
+					return
```
```suggestion
			var/static/list/part_list = list("Ears", "Tails", "Frills", "Spines", "Wings") //Selection, if you want more stuff added, do it yourself
			var/selected_part = input(owner, "Which bodypart do you want to alter?", "Mutant Bodypart Alteration") as null|anything in part_list + "Cancel"
			if(!selected_part || selected_part == "Cancel")
				return
```

Kapu1178 commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -972,6 +972,9 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 					S = GLOB.kepori_body_feathers_list[H.dna.features["kepori_body_feathers"]]
 			if(!S || S.icon_state == "none")
 				continue
+			S.image_alpha = H.dna.species.hair_alpha //hair_alpha is a decent indicator of how transparent a species is
+			if(isjellyperson(H))
+				S.hasinner = FALSE //Jellies shouldn't get inner sprites, where would they get the second color from?
```
this is still snowflakey and awful.

Kapu1178 commented on code/modules/mob/dead/new_player/sprite_accessories.dm:
```diff
@@ -1792,7 +1792,7 @@
 	color_src = HAIR
 
-/datum/sprite_accessory/ears/cat/slime
+/datum/sprite_accessory/ears/cat/slime //I would yeet these, but unfortunately, deleting these can mess up savefiles, so you're stuck with this :)
```
You can remove these. im pretty sure. Ask mark.

MarkSuckerberg replied:
Yeah, it'll just sanitize it and re-randomize if needed I believe.

Kapu1178 commented on code/modules/mob/living/carbon/human/species_types/jellypeople.dm:
```diff
@@ -171,44 +171,94 @@
+					if("Spines")
+						handle_sprites(lowertext(selected_part), H, GLOB.spines_list)
+					if("Wings") //These don't look too good
+						handle_sprites(lowertext(selected_part), H, GLOB.wings_list)
+	return
```
Just dump an `update_body(H)` before the return and remove all the other update_body() calls

Kapu1178 reviewed (changes requested):
These in addition to what was discussed on discord
